lib/active_merchant/billing/credit_card_methods.rb

```diff
@@ -11,7 +11,7 @@ module CreditCardMethods
         'american_express'   => ->(num) { num =~ /^3[47]\d{13}$/ },
         'naranja'            => ->(num) { num&.size == 16 && in_bin_range?(num.slice(0, 6), NARANJA_RANGES) },
         'diners_club'        => ->(num) { num =~ /^3(0[0-5]|[68]\d)\d{11,16}$/ },
-        'jcb'                => ->(num) { num =~ /^35(28|29|[3-8]\d)\d{12}$/ },
+        'jcb'                => ->(num) { num =~ /^35(28|29|[3-8]\d)\d{12}$|^308800\d{10}$/ },
```

Comment:
Should the first `$` still be in there now that it was added in towards the end?

Reply:
Well, I was thinking this regex was providing two completely separate options, so I needed the beginning-of-line operator and end-of-line operator in there twice, once for each option.

But based on your comment I looked back through the file for similar patterns and realized that it could be simplified as
`/^(35(28|29|[3-8]\d)\d{12}|308800\d{10})$/`

...which is probably better. So I will change it. Thank you for pointing that out.

Reply:
The `|` symbol represents an `or` condition for the regex, so it should be the first pattern or the second. The `$` just denotes "end of line" for the pattern we are expecting, meaning the regular expression should match from `^` (beggining) to `$` (end).